tests/rules/javascript/test.event_listener_fourth.js

```diff
@@ -5,129 +5,109 @@ import { VALIDATION_WARNING } from 'const';
   });
 
-  it('should allow a falsy literal', () => {
+  it('should allow a falsy literal', async () => {
     const code = 'window.addEventListener("click", function(){}, false, "");';
     const jsScanner = new JavaScriptScanner(code, 'badcode.js');
 
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(0);
-      });
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
   });
 
-  it('should not allow a true identifier', () => {
+  it('should not allow a true identifier', async () => {
     const code = oneLine`var t = 'true';
       window.addEventListener("click", function(){}, false, t);`;
     const jsScanner = new JavaScriptScanner(code, 'badcode.js');
 
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(1);
-        expect(linterMessages[0].code).toEqual(
-          messages.EVENT_LISTENER_FOURTH.code
-        );
-        expect(linterMessages[0].type).toEqual(VALIDATION_WARNING);
-      });
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(1);
+    expect(linterMessages[0].code).toEqual(
+      messages.EVENT_LISTENER_FOURTH.code
+    );
+    expect(linterMessages[0].type).toEqual(VALIDATION_WARNING);
   });
 
-  it('should allow a false identifier', () => {
+  it('should allow a false identifier', async () => {
     const code = oneLine`var t = false;
       window.addEventListener("click", function(){}, false, t);`;
     const jsScanner = new JavaScriptScanner(code, 'badcode.js');
 
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(0);
-      });
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
   });
 
-  it('should not allow a window identifier', () => {
+  it('should not allow a window identifier', async () => {
     const code = oneLine`var foo = window;
       foo.addEventListener("click", function(){}, false, false);`;
     const jsScanner = new JavaScriptScanner(code, 'badcode.js');
 
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(0);
-      });
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
   });
 
-  it('should not allow a hidden addEventListener identifier', () => {
+  it('should not allow a hidden addEventListener identifier', async () => {
     const code = oneLine`var foo = window.addEventListener;
       foo("click", function(){}, false, false);`;
     const jsScanner = new JavaScriptScanner(code, 'badcode.js');
 
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(0);
-      });
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
   });
```

Comment:
The question stays: Can't see where in the file event-listener-fourth.js we are checking and not allowing window identifier.
Also why 'should not allow test' doesn't report any warnings.